File: js/1043lies.py
import sys 
from collections import deque
input = sys.stdin.readline 

peopleCount, partyCount = map(int, input().split())
knowsTruth = [0] * peopleCount 
knowLine = list(map(int, input().split()))
needsVisit = deque() 

for i in range(knowLine.pop(0)):
    knowsTruth[knowLine[i]-1] = 1
    needsVisit.append(knowLine[i])

closeWeb = {i : set() for i in range(1, peopleCount+1)}

parties = [] # [attendMemberCount, attendingMembers]
for party in range(partyCount):
    members = list(map(int, input().split()))
    totalAttend = members.pop(0)
    parties.append([totalAttend, members])
    for person in members:
        closeWeb[person].update(members)

def bfs():
    visited = [False] * (peopleCount)
    while needsVisit:
        current = needsVisit.popleft()
        visited[current-1]=True
        knowsTruth[current-1] = 1

        for friend in closeWeb[current]:
            if not visited[friend-1]:
                knowsTruth[friend-1] = 1
                needsVisit.append(friend)

bfs()

answer = 0
for i in range(partyCount):

    for j in range(parties[i][0]):
        if knowsTruth[parties[i][1][j]-1]:
            break
    else:
        answer += 1


print(answer)

File: js/1043lies.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let peopleNum = 0;
let partyNum = [];
let knows = -1;
let knowsTruth = [];
let parties = [];
// [attendNum, membesr[], flag]

rl.on("line", function (line) {
  let members = line.split(" ").map((v) => parseInt(v));
  if (peopleNum === 0) {
    peopleNum = members[0];
    partyNum = members[1];
    knowsTruth = Array(peopleNum).fill(0);
  } else if (knows === -1) {
    knows = members.shift();
    members.map((mem) => (knowsTruth[mem - 1] = 1));
  } else {
    parties.push([members.shift(), members], true);
  }
});

const bfs = (person) => {
  let needsCheck = [];
  parties.map(({ attendNum, members, flag }) => {
    if (flag) {
      if (members.includes(person)) {
        needsCheck += members;
      }
    }
  });
};

rl.on("close", function () {
  console.log(knowsTruth);
  console.log(parties);
});
